sdk/openai/openai/MIGRATION.md

```diff
@@ -0,0 +1,498 @@
+- The `deploymentName` parameter is not needed if the client was created with the `deployment` option. If the client was not created with the `deployment` option, the `model` property in the option object should be set with the deployment name.
+
+### Streaming chat completions
+
+The following example shows how to migrate the `streamChatCompletions` method call.
+
+Original code:
+```typescript
+const stream = await client.streamChatCompletions(deploymentName, messages, { maxTokens: 100 });
+```
+
+Migrated code:
+```typescript
+const stream = await client.chat.completions.create({ messages, max_tokens: 100, stream: true });
+```
+
+### Azure On Your Data
+
+The following example shows how to migrate a `getChatCompletions` method call that enables Azure On Your Data with Azure Search.
+
+Original code:
+```typescript
+const azureSearchEndpoint = "Your Azure Search resource endpoint";
+const azureSearchIndexName = "Your Azure Search index name";
+const stream = await client.getChatCompletions(deploymentName, messages, { azureExtensionOptions: { 
+    extensions: [{
+      type: "azure_search",
+      endpoint: azureSearchEndpoint,
+      indexName: azureSearchIndexName,
+      authentication: {
+        type: "system_assigned_managed_identity",
+      }
+    }]
+  } 
+});
+```
+
+Migrated code:
+```typescript
+const azureSearchEndpoint = "Your Azure Search resource endpoint";
+const azureSearchIndexName = "Your Azure Search index name";
+const result = await client.chat.completions.create({ messages, ... { extensions: [{
+      type: "azure_search",
+      parameters: {
+        endpoint: azureSearchEndpoint,
+        index_name: azureSearchIndexName,
+        authentication: {
+          type: "system_assigned_managed_identity",
+        }
+      }
+    }]
+  } as any
+});
+```
+
+Notice that:
+- The `azureExtensionOptions` property has been replaced with the inner `extensions` property
+- The `parameters` property has been added to wrap the parameters of the extension, which mirrors the schema of the Azure OpenAI service API
+
+### Audio transcription
+
+The following example shows how to migrate the `getAudioTranscription` method call.
+
+Original code:
+```typescript
+import { readFile } from "fs/promises";
+
+const audioFilePath = "path/to/audio/file";
+const audio = await readFile(audioFilePath);
+const result = await client.getAudioTranscription(deploymentName, audio);
+```
+
+Migrated code:
+```typescript
+import { createReadStream } from "fs";
+
+const result = await client.audio.transcriptions.create({
+  model: '',
+  file: createReadStream(audioFilePath),
+});
+```
+
+Notice that:
+- The `getAudioTranscription` method has been replaced with the `audio.transcriptions.create` method
+- The `AzureOpenAI` has to be constructed with the `deployment` option set to the deployment name in order to use audio operations such as `audio.transcriptions.create`
```

Comment:
is this missing in the above snippet or am I misunderstanding the meaning?

Reply:
The examples for migrating methods don't repeat the code for client construction. The example for client construction is listed earlier in https://github.com/Azure/azure-sdk-for-js/blob/f4cac5d2858019b40170f309770583143fd7ab74/sdk/openai/openai/MIGRATION.md#constructing-the-client and it sets the deployment option.